src/Listener/PodeRequest.cs

```diff
@@ -292,108 +316,23 @@ public async Task<bool> Receive(CancellationToken cancellationToken)
                 PodeHelpers.WriteException(ex, Context.Listener, PodeLoggingLevel.Error);
                 Error = ex;
             }
+            catch (ObjectDisposedException ex)
+            {
+                PodeHelpers.WriteException(ex, Context.Listener, PodeLoggingLevel.Error);
+            }
             catch (Exception ex)
             {
                 PodeHelpers.WriteException(ex, Context.Listener, PodeLoggingLevel.Error);
                 Error = new PodeRequestException(ex, 500);
             }
             finally
             {
+                StreamLock.Release();
                 PartialDispose();
             }
-
             return false;
         }
 
-        /// <summary>
-        /// Reads data from the input stream until the specified bytes are found.
-        /// </summary>
-        /// <param name="checkBytes">The bytes to check for in the input stream.</param>
-        /// <param name="cancellationToken">Token to monitor for cancellation requests.</param>
-        /// <returns>A Task representing the async operation, with a string containing the data read.</returns>
-        public async Task<string> Read(byte[] checkBytes, CancellationToken cancellationToken)
```

Comment:
Could we re-add this `Read` method, and the `ValidateInputInternal` method below please? I know Read isn't used currently, but I will need it eventually for another feature I'm looking to add.